src/main/java/io/airlift/slice/SizeOf.java

```diff
@@ -97,6 +105,26 @@ public static long sizeOf(Object[] array)
         return (array == null) ? 0 : sizeOfObjectArray(array.length);
     }
 
+    public static <T> long sizeOf(Optional<T> optional, ToLongFunction<T> valueSize)
+    {
+        return optional.isPresent() ? OPTIONAL_INSTANCE_SIZE + valueSize.applyAsLong(optional.get()) : 0;
+    }
+
+    public static <T> long sizeOf(OptionalInt optional)
+    {
+        return optional.isPresent() ? OPTIONAL_INT_INSTANCE_SIZE : 0;
```

Comment:
Why 0 if it's absent? Objects of type OptionalInt still occupy space in that case.